test/integration/cdi/example-driver/kubeletplugin/noderegistrar.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package app does all of the work necessary to configure and run a
+// Kubernetes app process.
+package kubeletplugin
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"os/signal"
+	"runtime"
+	"syscall"
+
+	"golang.org/x/sys/unix"
+	"google.golang.org/grpc"
+	"k8s.io/klog/v2"
+	registerapi "k8s.io/kubelet/pkg/apis/pluginregistration/v1"
+)
+
+type nodeRegistrarConfig struct {
+	draDriverName          string
+	draAddress             string
+	pluginRegistrationPath string
+}
+type nodeRegistrar struct {
+	config nodeRegistrarConfig
+}
+
+func buildSocketPath(cdiDriverName string, pluginRegistrationPath string) string {
+	return fmt.Sprintf("%s/%s-reg.sock", pluginRegistrationPath, cdiDriverName)
+}
+
+func removeRegSocket(cdiDriverName, pluginRegistrationPath string) {
+	sigc := make(chan os.Signal, 1)
+	signal.Notify(sigc, syscall.SIGTERM)
+	<-sigc
+	socketPath := buildSocketPath(cdiDriverName, pluginRegistrationPath)
+	err := os.Remove(socketPath)
+	if err != nil && !os.IsNotExist(err) {
+		klog.Errorf("failed to remove socket: %s with error: %+v", socketPath, err)
+		os.Exit(1)
+	}
+	os.Exit(0)
+}
+
+func umask(mask int) (int, error) {
+	return unix.Umask(mask), nil
```

Comment:
Why do you write a wrapper function that returns a nil error when below you then ignore the error?

Reply:
Stupid code, let me just remove that function!